assets/js/base/context/providers/cart-checkout/checkout-state/reducer.ts

```diff
@@ -0,0 +1,195 @@
+ * @param {boolean} action.shouldCreateAccount True if shopper has requested a user account (sign-up checkbox).
+ * @param {Object} action.data Other action payload.
+ */
+export const reducer = (
+	state = DEFAULT_STATE,
+	{
+		url,
+		type,
+		customerId,
+		orderId,
+		orderNotes,
+		shouldCreateAccount,
+		data,
+	}: ActionType
+): CheckoutStateContextState => {
+	let newState = state;
+	switch ( type ) {
+		case ACTION.SET_PRISTINE:
+			newState = DEFAULT_STATE;
+			break;
+		case ACTION.SET_IDLE:
+			newState =
+				state.status !== STATUS.IDLE
+					? {
+							...state,
+							status: STATUS.IDLE,
+					  }
+					: state;
+			break;
+		case ACTION.SET_REDIRECT_URL:
+			newState =
+				url !== undefined && url !== state.redirectUrl
+					? {
+							...state,
+							redirectUrl: url,
+					  }
+					: state;
+			break;
+		case ACTION.SET_PROCESSING_RESPONSE:
+			newState = {
+				...state,
+				processingResponse: ( data as PaymentResultDataType ) || null,
+			};
+			break;
+
+		case ACTION.SET_COMPLETE:
+			newState =
+				state.status !== STATUS.COMPLETE
+					? {
+							...state,
+							status: STATUS.COMPLETE,
+							redirectUrl:
+								data !== undefined &&
+								data.redirectUrl !== undefined
+									? ( data.redirectUrl as string )
+									: state.redirectUrl,
+					  }
+					: state;
+			break;
+		case ACTION.SET_PROCESSING:
+			newState =
+				state.status !== STATUS.PROCESSING
+					? {
+							...state,
+							status: STATUS.PROCESSING,
+							hasError: false,
+					  }
+					: state;
+			// clear any error state.
+			newState =
+				newState.hasError === false
+					? newState
+					: { ...newState, hasError: false };
+			break;
+		case ACTION.SET_BEFORE_PROCESSING:
+			newState =
+				state.status !== STATUS.BEFORE_PROCESSING
+					? {
+							...state,
+							status: STATUS.BEFORE_PROCESSING,
+							hasError: false,
+					  }
+					: state;
+			break;
+		case ACTION.SET_AFTER_PROCESSING:
+			newState =
+				state.status !== STATUS.AFTER_PROCESSING
+					? {
+							...state,
+							status: STATUS.AFTER_PROCESSING,
+					  }
+					: state;
+			break;
+		case ACTION.SET_HAS_ERROR:
+			newState = state.hasError
+				? state
+				: {
+						...state,
+						hasError: true,
+				  };
+			newState =
+				state.status === STATUS.PROCESSING ||
+				state.status === STATUS.BEFORE_PROCESSING
+					? {
+							...newState,
+							status: STATUS.IDLE,
+					  }
+					: newState;
+			break;
+		case ACTION.SET_NO_ERROR:
+			newState = state.hasError
+				? {
+						...state,
+						hasError: false,
+				  }
+				: state;
+			break;
+		case ACTION.INCREMENT_CALCULATING:
+			newState = {
+				...state,
+				calculatingCount: state.calculatingCount + 1,
+			};
+			break;
+		case ACTION.DECREMENT_CALCULATING:
+			newState = {
+				...state,
+				calculatingCount: Math.max( 0, state.calculatingCount - 1 ),
+			};
+			break;
+		case ACTION.SET_CUSTOMER_ID:
+			newState =
+				customerId !== undefined
```

Comment:
There are not, no. The usage we have comes from the API and in those cases, they return `0` rather than undefined. `0` would be correct for both the customer ID and order ID if there are none or the user is a guest. We should enforce the prop.